circleci: failed tests: TestBasicManualReplication

The following test appears to have failed:

[#22221](https://circleci.com/gh/cockroachdb/cockroach/22221):

```
I160902 18:12:47.896512 634 rpc/context.go:169  closing 127.0.0.1:52390
I160902 18:12:47.896999 689 /go/src/google.golang.org/grpc/clientconn.go:767  grpc: addrConn.transportMonitor exits due to: context canceled
I160902 18:12:47.897670 706 rpc/context.go:169  closing 127.0.0.1:47391
I160902 18:12:47.897889 755 rpc/context.go:169  closing 127.0.0.1:52390
I160902 18:12:47.899623 472 storage/engine/rocksdb.go:402  closing in-memory rocksdb instance
--- FAIL: TestBasicManualReplication (6.24s)
	testcluster_test.go:151: change replicas of range 1 failed: context deadline exceeded
=== RUN   TestWaitForFullReplication
I160902 18:12:47.905077 838 storage/engine/rocksdb.go:354  opening in memory rocksdb instance
I160902 18:12:47.927630 838 gossip/gossip.go:230  [n?] initial resolvers: []
W160902 18:12:47.927764 838 gossip/gossip.go:1041  [n?] no resolvers found; use --join to specify a connected node
W160902 18:12:47.932886 838 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I160902 18:12:47.939308 838 server/node.go:413  [n?] store store=0:0 not bootstrapped
I160902 18:12:47.949639 871 storage/replica.go:2237  [s1,r1{/Min-/Max}] applying command with forced error: storage/replica.go:2162: no-op on empty Raft entry
E160902 18:12:47.950437 871 storage/replica.go:2272  [s1,r1{/Min-/Max}] error executing raft command: storage/replica.go:2162: no-op on empty Raft entry
I160902 18:12:47.951284 871 storage/replica_trigger.go:308  [s1,r1{/Min-/Max}] new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 409122h12m56.947837084s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0s [physicalTime=2016-09-02 18:12:47.951066635 +0000 UTC]
--
I160902 18:12:59.727160 1669 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:38792->127.0.0.1:53046: use of closed network connection
I160902 18:12:59.727425 1566 /go/src/google.golang.org/grpc/clientconn.go:767  grpc: addrConn.transportMonitor exits due to: context canceled
I160902 18:12:59.727599 1691 /go/src/google.golang.org/grpc/clientconn.go:767  grpc: addrConn.transportMonitor exits due to: context canceled
I160902 18:12:59.728078 1347 storage/engine/rocksdb.go:402  closing in-memory rocksdb instance
--- PASS: TestStopServer (5.04s)
FAIL
FAIL	github.com/cockroachdb/cockroach/testutils/testcluster	22.988s
=== RUN   TestStoreTimeSeries
I160902 18:13:13.885063 23 gossip/gossip.go:230  initial resolvers: []
W160902 18:13:13.885951 23 gossip/gossip.go:1041  no resolvers found; use --join to specify a connected node
I160902 18:13:13.886734 23 storage/engine/rocksdb.go:354  opening in memory rocksdb instance
I160902 18:13:13.900643 23 gossip/gossip.go:264  NodeID set to 1
I160902 18:13:13.904550 33 storage/replica.go:2845  [s1,r1{/Min-/Max}] gossiping cluster id {"00000000-0000-0000-0000-000000000000"} from store 1, range 1
I160902 18:13:13.905180 23 storage/store.go:975  store=1:1: failed initial metrics computation: store=1:1: system config not yet available
I160902 18:13:13.906857 23 gossip/gossip.go:264  NodeID set to 1
I160902 18:13:13.907285 23 gossip/gossip.go:278  NodeDescriptor set to node_id:1 address:<network_field:"" address_field:"" > attrs:<> 
I160902 18:13:13.915498 6 storage/replica.go:2237  [s1,r1{/Min-/Max}] applying command with forced error: storage/replica.go:2162: no-op on empty Raft entry

```
Please assign, take a look and update the issue accordingly.

--- a/kv/dist_sender.go
+++ b/kv/dist_sender.go
@@ -368,9 +368,8 @@ func (ds *DistSender) sendRPC(
 
 	// Set RPC opts with stipulation that one of N RPCs must succeed.
 	rpcOpts := SendOptions{
+		ctx:              ctx,
 		SendNextTimeout:  ds.sendNextTimeout,
-		Timeout:          base.NetworkTimeout,
-		Context:          ctx,
 		transportFactory: ds.transportFactory,
 	}
 	tracing.AnnotateTrace()
@@ -1047,7 +1046,7 @@ func (ds *DistSender) sendToReplicas(
 			sendNextTimer.Read = true
 			// On successive RPC timeouts, send to additional replicas if available.
 			if !transport.IsExhausted() {
-				log.Trace(opts.Context, "timeout, trying next peer")
+				log.Trace(opts.ctx, "timeout, trying next peer")
 				pending++
 				transport.SendNext(done)
 			}
@@ -1057,9 +1056,9 @@ func (ds *DistSender) sendToReplicas(
 			err := call.Err
 			if err == nil {
 				if log.V(2) {
-					log.Infof(opts.Context, "RPC reply: %+v", call.Reply)
+					log.Infof(opts.ctx, "RPC reply: %+v", call.Reply)
 				} else if log.V(1) && call.Reply.Error != nil {
-					log.Infof(opts.Context, "application error: %s", call.Reply.Error)
+					log.Infof(opts.ctx, "application error: %s", call.Reply.Error)
 				}
 
 				if !ds.handlePerReplicaError(rangeID, call.Reply.Error) {
@@ -1075,12 +1074,12 @@ func (ds *DistSender) sendToReplicas(
 				// information than a RangeNotFound).
 				err = call.Reply.Error.GoError()
 			} else if log.V(1) {
-				log.Warningf(opts.Context, "RPC error: %s", err)
+				log.Warningf(opts.ctx, "RPC error: %s", err)
 			}
 
 			// Send to additional replicas if available.
 			if !transport.IsExhausted() {
-				log.Tracef(opts.Context, "error, trying next peer: %s", err)
+				log.Tracef(opts.ctx, "error, trying next peer: %s", err)
 				pending++
 				transport.SendNext(done)
 			}

--- a/kv/send_test.go
+++ b/kv/send_test.go
@@ -20,7 +20,6 @@ import (
 	"net"
 	"reflect"
 	"strconv"
-	"strings"
 	"testing"
 	"time"
 
@@ -80,7 +79,7 @@ func TestInvalidAddrLength(t *testing.T) {
 
 	// The provided replicas is nil, so its length will be always less than the
 	// specified response number
-	opts := SendOptions{Context: context.Background()}
+	opts := SendOptions{ctx: context.Background()}
 	ds := &DistSender{Ctx: context.Background()}
 	ret, err := ds.sendToReplicas(opts, 0, nil, roachpb.BatchRequest{}, nil)
 
@@ -102,11 +101,7 @@ func TestSendToOneClient(t *testing.T) {
 	s, ln := newTestServer(t, ctx)
 	roachpb.RegisterInternalServer(s, Node(0))
 
-	opts := SendOptions{
-		SendNextTimeout: 1 * time.Second,
-		Timeout:         10 * time.Second,
-		Context:         context.Background(),
-	}
+	opts := SendOptions{ctx: context.Background()}
 	reply, err := sendBatch(opts, []net.Addr{ln.Addr()}, ctx)
 	if err != nil {
 		t.Fatal(err)
@@ -151,11 +146,7 @@ func TestRetryableError(t *testing.T) {
 		return nil
 	})
 
-	opts := SendOptions{
-		SendNextTimeout: 100 * time.Millisecond,
-		Timeout:         100 * time.Millisecond,
-		Context:         context.Background(),
-	}
+	opts := SendOptions{ctx: context.Background()}
 	if _, err := sendBatch(opts, []net.Addr{ln.Addr()}, clientContext); err == nil {
 		t.Fatalf("Unexpected success")
 	}
@@ -203,9 +194,8 @@ func setupSendNextTest(t *testing.T) ([]chan<- BatchCall, chan BatchCall, *stop.
 	doneChanChan := make(chan chan<- BatchCall, len(addrs))
 
 	opts := SendOptions{
+		ctx:             context.Background(),
 		SendNextTimeout: 1 * time.Millisecond,
-		Timeout:         10 * time.Second,
-		Context:         context.Background(),
 		transportFactory: func(_ SendOptions,
 			_ *rpc.Context,
 			replicas ReplicaSlice,
@@ -408,110 +398,6 @@ func TestSendNext_NonRetryableApplicationError(t *testing.T) {
 	}
 }
 
-// TestClientNotReady verifies that Send gets an RPC error when a client
-// does not become ready.
-func TestClientNotReady(t *testing.T) {
-	defer leaktest.AfterTest(t)()
-
-	stopper := stop.NewStopper()
-	defer stopper.Stop()
-
-	// Construct a server that listens but doesn't do anything. Note that we
-	// don't accept any connections on the listener.
-	ln, err := net.Listen(util.TestAddr.Network(), util.TestAddr.String())
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer ln.Close()
-	addr := ln.Addr()
-	addrs := []net.Addr{addr}
-
-	{
-		// Send RPC to an address where no server is running.
-		nodeContext := newNodeTestContext(nil, stopper)
-		if _, err := sendBatch(SendOptions{
-			SendNextTimeout: 100 * time.Nanosecond,
-			Timeout:         100 * time.Nanosecond,
-			Context:         context.Background(),
-		}, addrs, nodeContext); !testutils.IsError(err, "context deadline exceeded") {
-			t.Fatalf("unexpected error: %v", err)
-		}
-
-		// Do a dance to convince GRPC to close the connection.
-		if conn, err := ln.Accept(); err != nil {
-			t.Fatal(err)
-		} else {
-			// The connection is cached in the RPC context.
-			grpcConn, err := nodeContext.GRPCDial(addr.String())
-			if err != nil {
-				t.Fatal(err)
-			}
-			if err := grpcConn.Close(); err != nil && err != grpc.ErrClientConnClosing {
-				t.Fatal(err)
-			}
-
-			// Just in case, close it from the server as well.
-			if err := conn.Close(); err != nil {
-				t.Fatal(err)
-			}
-		}
-	}
-
-	errCh := make(chan error)
-	connected := make(chan struct{})
-
-	// Accept a single connection from the client.
-	go func() {
-		if _, err := ln.Accept(); err != nil {
-			errCh <- err
-		} else {
-			close(connected)
-		}
-	}()
-
-	// Send the RPC again with no timeout. We create a new node context to ensure
-	// there is a new connection; we could reuse the old connection by not
-	// closing it, but by the time we reach this point in the test, GRPC may have
-	// attempted to reconnect enough times to make the backoff long enough to
-	// time out the test.
-	nodeContext := newNodeTestContext(nil, stopper)
-
-	go func() {
-		_, err := sendBatch(SendOptions{
-			Context: context.Background(),
-		}, addrs, nodeContext)
-		expected := strings.Join([]string{
-			"context canceled",
-			"connection is closing",
-			"failed fast due to transport failure",
-		}, "|")
-		if !testutils.IsError(err, expected) {
-			errCh <- errors.Wrap(err, "unexpected error")
-		} else {
-			close(errCh)
-		}
-	}()
-
-	select {
-	case err := <-errCh:
-		t.Fatalf("unexpected error: %v", err)
-	case <-connected:
-	}
-
-	// Grab the cached connection and close it. This will cause the blocked RPC
-	// to finish.
-	grpcConn, err := nodeContext.GRPCDial(addr.String())
-	if err != nil {
-		t.Fatal(err)
-	}
-	if err := grpcConn.Close(); err != nil && err != grpc.ErrClientConnClosing {
-		t.Fatal(err)
-	}
-	for err := range errCh {
-		t.Fatal(err)
-	}
-}
-
 // firstNErrorTransport is a mock transport that sends an error on
 // requests to the first N addresses, then succeeds.
 type firstNErrorTransport struct {
@@ -577,9 +463,7 @@ func TestComplexScenarios(t *testing.T) {
 		}
 
 		opts := SendOptions{
-			SendNextTimeout: 1 * time.Second,
-			Timeout:         10 * time.Second,
-			Context:         context.Background(),
+			ctx: context.Background(),
 			transportFactory: func(_ SendOptions,
 				_ *rpc.Context,
 				replicas ReplicaSlice,

--- a/kv/transport.go
+++ b/kv/transport.go
@@ -40,22 +40,13 @@ var enableLocalCalls = envutil.EnvOrDefaultBool("COCKROACH_ENABLE_LOCAL_CALLS", 
 // more replicas, depending on error conditions and how many successful
 // responses are required.
 type SendOptions struct {
-	context.Context // must not be nil
+	ctx context.Context
+
 	// SendNextTimeout is the duration after which RPCs are sent to
 	// other replicas in a set.
 	SendNextTimeout time.Duration
-	// Timeout is the maximum duration of an RPC before failure.
-	// 0 for no timeout.
-	Timeout time.Duration
 
 	transportFactory TransportFactory
-}
-
-func (so SendOptions) contextWithTimeout() (context.Context, func()) {
-	if so.Timeout != 0 {
-		return context.WithTimeout(so.Context, so.Timeout)
-	}
-	return so.Context, func() {}
 }
 
 type batchClient struct {
@@ -165,13 +156,10 @@ func (gt *grpcTransport) SendNext(done chan<- BatchCall) {
 
 	addr := client.remoteAddr
 	if log.V(2) {
-		log.Infof(gt.opts.Context, "sending request to %s: %+v", addr, client.args)
+		log.Infof(gt.opts.ctx, "sending request to %s: %+v", addr, client.args)
 	}
 
 	if localServer := gt.rpcContext.GetLocalInternalServerForAddr(addr); enableLocalCalls && localServer != nil {
-		ctx, cancel := gt.opts.contextWithTimeout()
-		defer cancel()
-
 		// Clone the request. At the time of writing, Replica may mutate it
 		// during command execution which can lead to data races.
 		//
@@ -184,20 +172,17 @@ func (gt *grpcTransport) SendNext(done chan<- BatchCall) {
 			client.args.Txn = &clonedTxn
 		}
 
-		reply, err := localServer.Batch(ctx, &client.args)
+		reply, err := localServer.Batch(gt.opts.ctx, &client.args)
 		done <- BatchCall{Reply: reply, Err: err}
 		return
 	}
 
 	go func() {
-		ctx, cancel := gt.opts.contextWithTimeout()
-		defer cancel()
-
-		reply, err := client.client.Batch(ctx, &client.args)
+		reply, err := client.client.Batch(gt.opts.ctx, &client.args)
 		if reply != nil {
 			for i := range reply.Responses {
 				if err := reply.Responses[i].GetInner().Verify(client.args.Requests[i].GetInner()); err != nil {
-					log.Error(ctx, err)
+					log.Error(gt.opts.ctx, err)
 				}
 			}
 		}

--- a/sql/schema_changer_test.go
+++ b/sql/schema_changer_test.go
@@ -25,6 +25,8 @@ import (
 	"testing"
 	"time"
 
+	"golang.org/x/net/context"
+
 	"github.com/cockroachdb/cockroach/base"
 	"github.com/cockroachdb/cockroach/internal/client"
 	"github.com/cockroachdb/cockroach/keys"
@@ -430,7 +432,7 @@ ALTER INDEX t.test@foo RENAME TO ufo
 	}
 }
 
-func runSqlWithRetry(t *testing.T, db *gosql.DB, sql string) {
+func runSQLWithRetry(t *testing.T, db *gosql.DB, sql string) {
 	for {
 		_, err := db.Exec(sql)
 		if err == nil {
@@ -485,31 +487,31 @@ func runSchemaChangeWithOperations(
 	for i := 0; i < 10; i++ {
 		k := rand.Intn(maxValue)
 		v := maxValue + i + 1
-		runSqlWithRetry(t, sqlDB, fmt.Sprintf(`UPDATE t.test SET v = %d WHERE k = %d`, v, k))
+		runSQLWithRetry(t, sqlDB, fmt.Sprintf(`UPDATE t.test SET v = %d WHERE k = %d`, v, k))
 		updatedKeys = append(updatedKeys, k)
 	}
 
 	// Reupdate updated values back to what they were before.
 	for _, k := range updatedKeys {
-		runSqlWithRetry(t, sqlDB, fmt.Sprintf(`UPDATE t.test SET v = %d WHERE k = %d`, maxValue-k, k))
+		runSQLWithRetry(t, sqlDB, fmt.Sprintf(`UPDATE t.test SET v = %d WHERE k = %d`, maxValue-k, k))
 	}
 
 	// Delete some rows.
 	deleteStartKey := rand.Intn(maxValue - 10)
 	for i := 0; i < 10; i++ {
-		runSqlWithRetry(t, sqlDB, fmt.Sprintf(`DELETE FROM t.test WHERE k = %d`, deleteStartKey+i))
+		runSQLWithRetry(t, sqlDB, fmt.Sprintf(`DELETE FROM t.test WHERE k = %d`, deleteStartKey+i))
 	}
 	// Reinsert deleted rows.
 	for i := 0; i < 10; i++ {
 		k := deleteStartKey + i
-		runSqlWithRetry(t, sqlDB, fmt.Sprintf(`INSERT INTO t.test VALUES(%d, %d)`, k, maxValue-k))
+		runSQLWithRetry(t, sqlDB, fmt.Sprintf(`INSERT INTO t.test VALUES(%d, %d)`, k, maxValue-k))
 	}
 
 	// Insert some new rows.
 	numInserts := 10
 	for i := 0; i < numInserts; i++ {
 		k := maxValue + i + 1
-		runSqlWithRetry(t, sqlDB, fmt.Sprintf(`INSERT INTO t.test VALUES(%d, %d)`, k, k))
+		runSQLWithRetry(t, sqlDB, fmt.Sprintf(`INSERT INTO t.test VALUES(%d, %d)`, k, k))
 	}
 
 	wg.Wait() // for schema change to complete.
@@ -526,7 +528,7 @@ func runSchemaChangeWithOperations(
 
 	// Delete the rows inserted.
 	for i := 0; i < numInserts; i++ {
-		runSqlWithRetry(t, sqlDB, fmt.Sprintf(`DELETE FROM t.test WHERE k = %d`, maxValue+i+1))
+		runSQLWithRetry(t, sqlDB, fmt.Sprintf(`DELETE FROM t.test WHERE k = %d`, maxValue+i+1))
 	}
 }
 
@@ -718,7 +720,7 @@ func TestSchemaChangeRetry(t *testing.T) {
 				attempts++
 				// Return a deadline exceeded error once.
 				if attempts == 1 {
-					return errors.New("context deadline exceeded")
+					return context.DeadlineExceeded
 				}
 				return nil
 			},
@@ -799,7 +801,7 @@ func TestSchemaChangePurgeFailure(t *testing.T) {
 				// Return a deadline exceeded error during the second attempt
 				// which attempts to clean up the schema change.
 				if attempts == 2 {
-					return errors.New("context deadline exceeded")
+					return context.DeadlineExceeded
 				}
 				return nil
 			},

--- a/testutils/error.go
+++ b/testutils/error.go
@@ -55,7 +55,7 @@ func IsPError(pErr *roachpb.Error, re string) bool {
 // that show a connection issue or an issue with the node itself. This can
 // occur when a node is restarting or is unstable in some other way.
 func IsSQLRetryError(err error) bool {
-	return IsError(err, "(connection reset by peer|connection refused|failed to send RPC|EOF|context deadline exceeded)")
+	return IsError(err, "(connection reset by peer|connection refused|failed to send RPC|EOF)")
 }
 
 // Caller returns filename and line number info for the specified stack

